### PR TITLE
Detect IDevID/IAK template from certificates

### PR DESCRIPTION
**Pull request by Isaac-Matthews**
This adds the requested auto-detect functionality for the agent. The new default setting for the IDevID and IAK templates will be `detect`, and with that set the agent will detect what template has been used from the imported certificates. This can still be overridden by users that want to specify the template or algorithm but likely should be left as default for the majority of users.

The certs are now imported first, and the key regeneration is delayed until after, with the key comparison against the certs performed during key regeneration.

### CODE REVIEW

**Review by stefanberger — APPROVED** [review]

File: keylime-agent.conf [hunk 1]
```diff
@@ -228,19 +228,21 @@ tpm_signing_alg = "rsassa"
 ek_handle = "generate"
 
 # Enable IDevID and IAK usage and set their algorithms.
-# Choosing a template will override the name and asymmetric algorithm choices.
+# By default the template will be detected automatically from the certificates. This will happen in iak_idevid_template is left empty or set as "default" or "detect".
```
[the remark this answers is not in the document]
Comment (Isaac-Matthews) [reply keylime-agent.conf]: Thanks, fixed

**Comment by stefanberger** [comment]
Works with "detect" now. Very cool!

**Comment by ansasaki** [comment]
@Isaac-Matthews Thank you for implementing this! Could you please rebase?

**Comment by Isaac-Matthews** [comment]
> @Isaac-Matthews Thank you for implementing this! Could you please rebase?

No problem. Done!

**Review by ansasaki — APPROVED** [review]

**Pull request merged**